### ISSUE TITLE
Adding Android platform peer dependency errors

### ISSUE THREAD

**Issue by timstoute** [phiamo/capacitor-plugin-playlist#72]
I'm running:

   Capacitor CLI      : 4.2.0
   @capacitor/android : 4.2.0
   @capacitor/core    : 4.2.0
   @capacitor/ios     : 4.2.0

and trying to install the Capacitor Andorid platform with command: **ionic capacitor add android**

and getting this response with errors:

```> npm i -E @capacitor/android@4.2.0
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: capacitor-plugin-playlist@0.2.4
npm ERR! Found: @capacitor/core@4.2.0
npm ERR! node_modules/@capacitor/core
npm ERR!   peer @capacitor/core@"^4.0.0" from @capacitor/app@4.1.1
npm ERR!   node_modules/@capacitor/app
npm ERR!     @capacitor/app@"^4.0.1" from the root project
npm ERR!   peer @capacitor/core@"^4.0.0" from @capacitor/filesystem@4.1.1
npm ERR!   node_modules/@capacitor/filesystem
npm ERR!     @capacitor/filesystem@"4.1.1" from the root project
npm ERR!   6 more (@capacitor/ios, @capacitor/network, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @capacitor/core@"^3.0.0" from capacitor-plugin-playlist@0.2.4
npm ERR! node_modules/capacitor-plugin-playlist
npm ERR!   capacitor-plugin-playlist@"^0.2.4" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @capacitor/core@3.9.0
npm ERR! node_modules/@capacitor/core
npm ERR!   peer @capacitor/core@"^3.0.0" from capacitor-plugin-playlist@0.2.4
npm ERR!   node_modules/capacitor-plugin-playlist
npm ERR!     capacitor-plugin-playlist@"^0.2.4" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/xxxx/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/xxxx/.npm/_logs/2023-04-16T21_10_19_190Z-debug-0.log
[ERROR] An error occurred while running subprocess npm.
        
        npm i -E @capacitor/android@4.2.0 exited with exit code 1.
```

What can I do to work around these errors ?

**Comment by podbasico** [comment]
try passing this flag --legacy-peer-deps

It seems a conflict between the versions you have in your package.json
Can you paste it here?

**Comment by Gonzalo-Bruna** [comment]
Pretty sure it's because you're trying to use this plugin with capacitor v4 instead of v3 (this plugin was made for the v3 version), and therefore you have peer dependency errors

**Comment by phiamo** [comment]
should be fixed in 0.3.0
